### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by vektor8**

### CODE REVIEW

**Comment by vektor8** [comment]
Accidentally committed 3 times. Sorry for that

File: main.c [hunk 1]
```diff
@@ -1,32 +1,21 @@
 #include <stdio.h>
-#include <string.h>
+#include "userData&Order.h"
```
Comment (Ranapop) [inline main.c]: that '&' is a sign maybe you should split the functionality

File: printOptions.c [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    }
+    printf("d)Go back\n>");
+}
+
+void printDrinkOptions(char foodOptions[][MAX_FOOD_NAME], int foodsChoice, int nrOfDrinks, int drinkPrices[], char drink[][MAX_DRINK_NAME]) {
+    printf("Please choose a drink to go with your %s:\n", foodOptions[foodsChoice]);
+    for (int i = 1; i <= nrOfDrinks; i++) {
+        putchar('a' + i - 1);
+        printf(") %s", drink[i - 1]);
+        if (drinkPrices[i - 1] > 0) printf(" %d\n", drinkPrices[i - 1]);
```
Comment (Ranapop) [inline printOptions.c]: what are you doing here? looks avoidable

File: userData&Order.c [hunk 3]
```diff
@@ -0,0 +1,36 @@
+//
+// Created by Victor on 11/3/2019.
+//
+#include <stdio.h>
+#include <string.h>
+#define MAX_FOOD_NAME 10
```
Comment (Ranapop) [inline userData&Order.c]: you have them defined in multiple files. You could extract them in a file constants.h or something...

File: userData&Order.c [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    printf("Welcome to Food Thingies!\n");
+    printf("Please sign in to continue\n");
+    printf("---Username:\n>");
+    gets(Username);
+    printf("---Password:\n>");
+    gets(Password);
+}
+
+void printUserOrder(char Username[], char foodOptions[][MAX_FOOD_NAME], int foodsChoice, char foods[MAX_NR_FOODS][MAX_NR_FOODS][MAX_SPECIALITY_NAME],
+                    int specialityChoice, char YesNo[][4], int prices[MAX_NR_FOODS][MAX_NR_OF_PRICES], char drink[][MAX_DRINK_NAME],
```
Comment (Ranapop) [inline userData&Order.c]: you are sending more data then needed... you don't have to send the whole food options, just the chosen food.. same applies for other parameters. If you send only the data you need, you might also have less parameters (cause you're having quite a lot now)

**Review by Ranapop — CHANGES_REQUESTED** [review]
See code comments

**Comment by vektor8** [comment]
-split the functionality 
-stored all the constants in a file
-got rid of unnecessary parameters in functions

File: main.c [hunk 5]
```diff
@@ -1,99 +1,88 @@
 #include <stdio.h>
-#include "userData&Order.h"
+#include "getUserData.h"
```
Comment (Ranapop) [inline main.c]: would not start a file name with a verb. File names are usually nouns

File: main.c [hunk 6]
```diff
@@ -1,99 +1,88 @@
...
 #include "printOptions.h"
+#include "printOrder.h"
```
Comment (Ranapop) [inline main.c]: same here

File: main.c [hunk 7]
```diff
@@ -1,99 +1,88 @@
 #include <stdio.h>
-#include "userData&Order.h"
+#include "getUserData.h"
 #include "printOptions.h"
```
Comment (Ranapop) [inline main.c]: same here (use noun)

File: main.c [hunk 8]
```diff
@@ -1,99 +1,88 @@
...
+#include "printOrder.h"
+#include "constants.h"
 
 int getChoiceIndex();
 
+int cutleryChoice;
```
Comment (Ranapop) [inline main.c]: why global?

File: main.c [hunk 9]
```diff
@@ -1,99 +1,88 @@
...
     Intro:
        signIn(Username, Password);
-
-    Food:
-    {
-        printFoodOptions(nrOfFoods, foodOptions);
+    Food: {
+    printFoodOptions(nrOfFoods,foodOptions);
         foodsChoice=getChoiceIndex();
         if (foodsChoice == 3)
```
Comment (Ranapop) [inline main.c]: what 3?

**Review by Ranapop — CHANGES_REQUESTED** [review]
see code comments

**Comment by vektor8** [comment]
-renamed files in order for them to not contain verbs
-replaced "3" with a more general case: "nrOfFoods" and did this in multiple places
-variables are now declared in main and the main function has less than 60 lines of code

**Pull request merged**